Sources/DiagnosticsLogger.swift

```diff
@@ -198,39 +206,18 @@ private extension DiagnosticsLogger {
         // Send all output (STDOUT and STDERR) to our `Pipe`.
         dup2(inputPipe.fileHandleForWriting.fileDescriptor, STDOUT_FILENO)
         dup2(inputPipe.fileHandleForWriting.fileDescriptor, STDERR_FILENO)
-
-        // Observe notifications from our input `Pipe`.
-        NotificationCenter.default.addObserver(
-            self,
-            selector: #selector(handlePipeNotification(_:)),
-            name: FileHandle.readCompletionNotification,
-            object: pipeReadHandle
-        )
-
-        // Start asynchronously monitoring our `Pipe`.
-        pipeReadHandle.readInBackgroundAndNotify()
     }
 
-    @objc func handlePipeNotification(_ notification: Notification) {
-        defer {
-            // You have to call this again to continuously receive notifications.
-            inputPipe.fileHandleForReading.readInBackgroundAndNotify()
-        }
-
-        guard
-            let data = notification.userInfo?[NSFileHandleNotificationDataItem] as? Data,
-            let string = String(data: data, encoding: .utf8) else {
-                assertionFailure()
-                return
-        }
-
+    private func handleLoggedData(_ data: Data) {
         outputPipe.fileHandleForWriting.write(data)
 
-        queue.async {
-            string.enumerateLines(invoking: { (line, _) in
-                self.log("SYSTEM: \(line)\n")
-            })
+        guard let string = String(data: data, encoding: .utf8) else {
+            return assertionFailure()
```

Comment:
Could we put a message here that describes what is the cause of the failure?

Reply:
Yes, definitely!